### ISSUE TITLE
cmake 3.15 is showing warning about missing project statement

### ISSUE THREAD

**Issue by floooh** [floooh/fips#232]
```
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by floooh** [comment]
...so top-level CMakelists.txt files need an explicit project() command before calling fips_setup().

For backward compatibility, fips_setup() needs to check CMAKE_PROJECT_NAME, and if this already set, use this instead of setting the project-name again.

**Comment by floooh** [comment]
Fix is in d23e4d194e4879fd32ed78119ed76e20821eb4e7 (mainly documentation changes, and a warning in fips_setup())
